module/zcommon/zfs_fletcher.c

```diff
@@ -245,11 +214,258 @@ fletcher_4_incremental_byteswap(const void *buf, uint64_t size,
 	ZIO_SET_CHECKSUM(zcp, a, b, c, d);
 }
 
+static const struct fletcher_4_calls fletcher_4_generic_calls = {
+	.init = fletcher_4_generic_init,
+	.compute = fletcher_4_generic,
+	.compute_byteswap = fletcher_4_generic_byteswap,
+	.name = "generic"
+};
+
+extern struct fletcher_4_calls fletcher_4_avx2_calls;
+
+static const struct fletcher_4_calls *fletcher_4_algos[] = {
+	&fletcher_4_generic_calls,
+#if defined(HAVE_AVX) && defined(HAVE_AVX2)
+	&fletcher_4_avx2_calls,
+#endif
+};
+
+#define	ZFS_ARRAY_SIZE(a) 	(sizeof (a) / sizeof (*a))
```

Comment:
I've seen other patches request the same functionality, it probably makes sense to add this to `zfs_context.h`.  In kernel you can just defined ZFS_ARRAY_SIZE to ARRAY_SIZE, and then provide this version for user space.